components/brave_rewards/resources/android_page/components/pageWallet.tsx

```diff
@@ -49,17 +49,21 @@ class PageWallet extends React.Component<Props, State> {
     return utils.convertBalance(balance.total.toString(), balance.rates)
   }
 
-  getGrants = () => {
-    const grants = this.props.rewardsData.walletInfo.grants
-    if (!grants) {
+  generatePromotions = () => {
+    const promotions = this.props.rewardsData.promotions
+    if (!promotions) {
       return []
     }
 
-    return grants.map((grant: Rewards.Grant) => {
+    let claimedPromotions = promotions.filter((promotion: Rewards.Promotion) => {
+      return promotion.status === 4 // PromotionStatus::FINISHED
```

Comment:
nit: make a TODO or similar so it doesn't get overlooked - it's a big PR :)

Reply:
what do you mean with TODO?

Reply:
right now we can't use direct definition so I added reference for a number

Reply:
we use this approach all over the place. When we fix this problem we can migrate on all places

Reply:
not sure if needed